excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java

```diff
@@ -60,14 +60,39 @@ public static void register(ReaderRegistry registry) {
 
   @Override
   public Table read(XlsxReadOptions options) throws IOException {
-    List<Table> tables = readMultiple(options);
-    if (tables.isEmpty()) {
-      throw new IllegalArgumentException("No tables found.");
+    List<Table> tables = readMultiple(options, true);
+    if (options.sheetIndex() != null) {
+      int index = options.sheetIndex();
+      if (index < 0 || index >= tables.size()) {
+        throw new IndexOutOfBoundsException(
+            "Sheet index outside bounds. " + tables.size() + " sheets found.");
+      }
+
+      Table table = tables.get(index);
+      if (table == null) {
+        throw new IllegalArgumentException("No tables found.");
```

Comment:
Let's change this message to:

    throw new IllegalArgumentException("No table found at index " + index);

Reply:
Maybe something like `"No table found for sheet index " + index`?

Reply:
Or `found at sheet index`?

Reply:
sure. "at sheet index" sounds good to me